core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -164,24 +169,138 @@ public Boolean useExplicitFieldOrder() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("highlight");
+        builder.startObject(HIGHLIGHT_ELEMENT_NAME);
         innerXContent(builder);
         builder.endObject();
         return builder;
     }
 
+    /**
+     * Creates a new {@link HighlightBuilder} from the highlighter held by the {@link QueryParseContext}
+     * in {@link org.elasticsearch.common.xcontent.XContent} format
+     *
+     * @param parseContext
+     *            the input parse context. The state on the parser contained in
+     *            this context will be changed as a side effect of this method
+     *            call
+     * @return the new {@link HighlightBuilder}
+     */
+    public static HighlightBuilder fromXContent(QueryParseContext parseContext) throws IOException {
```

Comment:
I think this should be done as part of the highlighter refactoring as it will make the code more robust against inconsistencies. Whether it's in this PR of a subsequent one I will leave to you but I think it should be done soon